dash/dash-renderer/src/components/error/menu/VersionInfo.react.js

```diff
@@ -0,0 +1,182 @@
+import React, {useEffect, useState} from 'react';
+
+import './VersionInfo.css';
+
+const DAY_IN_MS = 86400000;
+
+function compareVersions(v1, v2) {
+    const v1Parts = v1.split('.').map(Number);
+    const v2Parts = v2.split('.').map(Number);
+
+    for (let i = 0; i < Math.max(v1Parts.length, v2Parts.length); i++) {
+        const part1 = v1Parts[i] || 0;
+        const part2 = v2Parts[i] || 0;
+
+        if (part1 > part2) return 1;
+        if (part1 < part2) return -1;
+    }
+
+    return 0;
+}
+
+async function requestDashVersionInfo(config) {
+    const {
+        dash_version: currentDashVersion,
+        dash_version_url: dashVersionUrl,
+        python_version: pythonVersion,
+        ddk_version: ddkVersion,
+        plotly_version: plotlyVersion
+    } = config;
+    const cachedVersionInfo = localStorage.getItem('cachedNewDashVersion');
+    const cachedNewDashVersionLink = localStorage.getItem(
+        'cachedNewDashVersionLink'
+    );
+    const lastFetched = localStorage.getItem('lastFetched');
+    if (
+        lastFetched &&
+        Date.now() - Number(lastFetched) < DAY_IN_MS &&
+        cachedVersionInfo
+    ) {
+        return {
+            version: JSON.parse(cachedVersionInfo),
+            link: cachedNewDashVersionLink
+        };
+    } else {
+        return fetch(dashVersionUrl, {
+            method: 'POST',
+            body: JSON.stringify({
+                dash_version: currentDashVersion,
+                python_version: pythonVersion,
+                ddk_version: ddkVersion,
+                plotly_version: plotlyVersion
+            }),
+            headers: {
+                'Content-Type': 'application/json'
+            }
+        })
+            .then(response => response.json())
+            .then(body => {
+                localStorage.setItem(
+                    'cachedNewDashVersion',
+                    JSON.stringify(body.version)
+                );
+                localStorage.setItem('cachedNewDashVersionLink', body.link);
+                localStorage.setItem('lastFetched', Date.now());
+                return body;
+            });
+    }
+}
+
+function shouldShowUpgradeNotification(currentDashVersion, newDashVersion) {
+    const showNotifications = localStorage.getItem('showNotifications');
+    const lastDismissed = localStorage.getItem('lastDismissed');
+    const lastDismissedVersion = localStorage.getItem('lastDismissedVersion');
+    if (
+        currentDashVersion == newDashVersion ||
```

Comment:
I could imagine edge cases where `currentDashVersion` is "greater" than `newDashVersion` -- maybe they are running a prerelease or custom build. 

Are the versions in a format where we can do greater-than/less-than comparisons?

```suggestion
        currentDashVersion >= newDashVersion ||
```